### PR TITLE
Improve memote.jl

### PR DESCRIPTION
**Pull request by stelmo**
Many many changes

### CODE REVIEW

**Comment by exaexa** [comment]
@stelmo pls nuke the big files from history; recommended way is to rebase -i and move the "removal" commit just below the commit that added them as a fixup.

File: test/memote.jl [hunk 1]
```diff
@@ -46,308 +46,274 @@ end
...
-    @test length(
-        FBCModelTests.Memote.Reaction.reactions_with_partially_identical_annotations(
-            negative_model,
-        ),
-    ) == 12
-
-    negative_model.reactions["SUCDi"].metabolites["succ_c"] = -2
-    @test isempty(FBCModelTests.Memote.Reaction.duplicate_reactions(negative_model))
-end
+# @testset "Consistency" begin
```
Comment (exaexa) [inline test/memote.jl:101]: I guess this code may be pwned completely as it is implicitly covered by the testcounting above, right?

Comment (stelmo) [reply test/memote.jl]: not sure what you mean. I have updated the tests now :)

**Comment by stelmo** [comment]
Nuked the big files

File: src/memote/checks/Reactions.jl [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import ..Config
+import ..Biomass: findall_biomass_reactions
+import ..GPRAssociation: reaction_has_sensible_gpr
+
+"""
+$(TYPEDSIGNATURES)
+
+Return a list of all reactions that are duplicated.
+"""
+function duplicate_reactions(model::MetabolicModel)
```
Comment (exaexa) [inline src/memote/checks/Reactions.jl:20]: naming rant: you're using mostly imperative functions (`find_xxxs(model)` actually means `xxxs_taken_from(model)`). Here I had hard time to guess whether this is making duplicates of the reactions (conforms with the rest of the naming) or is a predicate for 2 reactions (not enough parameters) or actually just lacks the usual `find_all_`. I'm actually in favor of the "functional" naming scheme as used here, without the imperatives. Perhaps better specify where you searching for the reactions (`duplicated_reactions_in_model`) or so.

No need to fix tho, this is a nitpick. :D

**Review by exaexa — DISMISSED** [review]
Few minor things to check/fix but basically OK.

File: src/memote/checks/Biomass.jl [hunk 3]
```diff
@@ -9,147 +9,112 @@ using DocStringExtensions
...
+
+    for (mid, v) in reaction_stoichiometry(model, rid)
+        annos = get(
+            parse_annotations(metabolite_annotations(model, mid)),
+            "bigg.metabolite",
+            String[],
+        ) # return safe fallback
+
+        # consume
+        if any(in(Config.atp, annos)) && v < 0
```
[the remark this answers is not in the document]
Comment (stelmo) [reply src/memote/checks/Biomass.jl:45]: ah no, you are right, any is unnecessary

[the remark this answers is not in the document]
Comment (stelmo) [reply src/memote/checks/Biomass.jl:45]: actually, I need the `in.`

File: src/memote/checks/Reactions.jl (hunk 2 above)
Comment (stelmo) [reply src/memote/checks/Reactions.jl:20]: fixing

**Pull request merged**